### DISCUSSION TITLE
Issues using Pyinstaller GitHub action

### DISCUSSION THREAD

**Discussion by NoelleTGS** [pyinstaller/pyinstaller#8886] [Help] [answered]
Hello, for months now I have been unable to use a particular Pyinstaller GitHub action with it crashing every time. It runs Pyinstaller through Wine to create a Windows executable for the program.

The error can be viewed here: https://github.com/NoelleTGS/osu-scorepost-generator/actions/runs/11669487563/job/33215076951
If it can't be viewed, I put it in a pastebin as well: https://pastebin.com/3s944s6V

I didn't report this for quite a while because I'm not sure whether the issue is with Pyinstaller, the action code, my own code, Wine, the package trying to be imported, etc... but was hoping I could at least be directed to the right location through here. Feel free to ask if any other information is needed.

**Comment by rokm** [comment, marked as answer]
Looks like that python sub-process in which we are importing packages before binary dependency analysis crashes because it ends up calling CRT function that is not implemented by Wine:

```
Unhandled exception: unimplemented function ucrtbase.dll.crealf called in 64-bit code (0x0000007b01390d).
```

So at its core, this is a Wine issue.

As per
```
PyInstaller.isolated._parent.SubprocessDiedError: Child process died calling import_library() with args=('circleguard',) and kwargs={}. Its exit code was 2147483904.
``` 

this happens when we try to import `circleguard`.

You can try to patch your copy of PyInstaller and add `circleguard`  to the `suppressed_imports` list at the end of this block:

https://github.com/pyinstaller/pyinstaller/blob/a2a6d1e5f1a8db5b7558b2158f2af21cb6a8e0b4/PyInstaller/building/build_main.py#L264-L279

That should skip importing `circleguard` at that stage of analysis, and might get you past this (particular instance of) issue.
